src/wallet/rpcdump.cpp

```diff
@@ -614,11 +611,10 @@ UniValue importwallet(const JSONRPCRequest& request)
                     }
                 }
                 pwallet->WalletLogPrintf("Importing %s...\n", EncodeDestination(keyid));
-                if (!pwallet->AddKeyPubKey(key, pubkey)) {
+                if (!pwallet->AddKeyPubKey(key, pubkey, nTime)) {
```

Comment:
I think this is actually a bugfix, because previously `mapKeyMetadata` got updated too late, after `CWallet::AddKeyPubKeyWithDB` and `CWallet::AddCryptedKey` already used the previous value:

https://github.com/bitcoin/bitcoin/blob/fb52d0684e0f03bca5d7ddcf3fc5b0657859ac2c/src/wallet/wallet.cpp#L242
https://github.com/bitcoin/bitcoin/blob/fb52d0684e0f03bca5d7ddcf3fc5b0657859ac2c/src/wallet/wallet.cpp#L267